### PR TITLE
Feature/ER-1016 provider location

### PR DESCRIPTION
**Pull request by shomavg**
Mostly copy & paste from the employer with minor changes in the way legal entity id is got.

### CODE REVIEW

File: src/Provider/Provider.Web/Orchestrators/Part1/LocationOrchestrator.cs [hunk 1]
```diff
@@ -117,64 +135,64 @@ public async Task<VacancyEmployerInfoModel> GetVacancyEmployerInfoModelAsync(Vac
...
             {
-                vacancy.EmployerLocation = new Vacancies.Client.Domain.Entities.Address
-                {
-                    AddressLine1 = selectedOrganisation.Address.AddressLine1,
-                    AddressLine2 = selectedOrganisation.Address.AddressLine2,
-                    AddressLine3 = selectedOrganisation.Address.AddressLine3,
-                    AddressLine4 = selectedOrganisation.Address.AddressLine4,
-                    Postcode = selectedOrganisation.Address.Postcode.AsPostcode()
-                };
+                vacancy.LegalEntityName = selectedOrganisation.Name;
```
Comment (dev4valtech) [inline src/Provider/Provider.Web/Orchestrators/Part1/LocationOrchestrator.cs]: This is happening twice. I would prefer this is exactly same as Employer, which has gone through enough testing and it is optimized.

Comment (shomavg) [reply src/Provider/Provider.Web/Orchestrators/Part1/LocationOrchestrator.cs]: it was copy paste error!

File: src/Provider/Provider.Web/Orchestrators/Part1/LocationOrchestrator.cs [hunk 2]
```diff
@@ -41,44 +43,55 @@ public class LocationOrchestrator : EntityValidatingOrchestrator<Vacancy, Locati
...
 
-            return await GetViewModelAsync(vacancy, employerInfoModel?.HasLegalEntityChanged, employerInfoModel?.LegalEntityId);
-        }
-        private async Task<LocationViewModel> GetViewModelAsync(Vacancy vacancy, bool? hasLegalEntityChanged, long? selectedOrganisationId)        
-        {
-            var employerData = await _providerVacancyClient.GetProviderEmployerVacancyDataAsync(vacancy.TrainingProvider.Ukprn.GetValueOrDefault(), vacancy.EmployerAccountId);
+            var vm = new LocationViewModel();
+            vm.PageInfo = Utility.GetPartOnePageInfo(vacancy);
+           
+            long? selectedOrganisationId = employerInfoModel?.LegalEntityId;
```
Comment (chrisjensenuk) [inline src/Provider/Provider.Web/Orchestrators/Part1/LocationOrchestrator.cs]: Can we copy the code from Employer.Web LocationOrchestrator? 
`var legalEntityId = employerInfoModel?.LegalEntityId != null ? employerInfoModel.LegalEntityId : vacancy.LegalEntityId;`

`selectedOrganisationId` 
 adds confusion.  We should standardise the code as much as possible between the applications.

File: src/Provider/Provider.Web/Orchestrators/Part1/LocationOrchestrator.cs [hunk 3]
```diff
@@ -41,44 +43,55 @@ public class LocationOrchestrator : EntityValidatingOrchestrator<Vacancy, Locati
...
                 vm.Review = await _reviewSummaryService.GetReviewSummaryViewModelAsync(vacancy.VacancyReference.Value,
                     ReviewFieldMappingLookups.GetLocationFieldIndicators());
             }
             return vm;
-        } 
+        }
+
+        private async Task<List<Address>> GetAllAvailableLocationsAsync(EmployerProfile employerProfile, Vacancy vacancy, long ukprn)
+        {
+            var employerData = await _providerVacancyClient.GetProviderEditVacancyInfoAsync(ukprn);
```
Comment (chrisjensenuk) [inline src/Provider/Provider.Web/Orchestrators/Part1/LocationOrchestrator.cs]: rename `employerData` to `providerData`

File: src/Provider/Provider.Web/Orchestrators/Part1/LocationOrchestrator.cs [hunk 4]
```diff
@@ -96,85 +109,81 @@ public async Task<VacancyEmployerInfoModel> GetVacancyEmployerInfoModelAsync(Vac
...
+            if (string.IsNullOrEmpty(locationEditModel.SelectedLocation))
+                return new OrchestratorResponse(false);
+
+            var vacancy = await Utility.GetAuthorisedVacancyForEditAsync(_providerVacancyClient,
+                _recruitVacancyClient, locationEditModel, RouteNames.Location_Post);
+
+            var employerVacancyInfo =
+                await _providerVacancyClient.GetProviderEmployerVacancyDataAsync(ukprn, vacancy.EmployerAccountId);
 
             LegalEntity selectedOrganisation;
```
Comment (chrisjensenuk) [inline src/Provider/Provider.Web/Orchestrators/Part1/LocationOrchestrator.cs]: Can we just do what Employer.Web does regarding the setting of `legalEntityId` & `selectedOrganisation` ? Its confusing having two ways of doing the same thing.

**Pull request merged**